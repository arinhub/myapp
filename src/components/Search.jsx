/* eslint-disable array-callback-return */
import React, {memo, useEffect, useState} from 'react';
import { useSelector, useDispatch, shallowEqual } from 'react-redux';
import {searchedItems, selectItems, removeSearchedItems} from '../actions';
import { ToastContainer, toast } from 'react-toastify';
const Search = (handleBack) => {
    const dispatch = useDispatch();
    const items= useSelector(selectItems, shallowEqual);
    const [value, setValue] = useState('');
    const getSearchQuery = (data) => {
        setValue(data)
        if (data.length === 0) dispatch(removeSearchedItems());
    };
    const searchData = () => {
        if (value.length === 0) return;
        let newArray = items.filter(el => el.name.match(new RegExp(value, 'gi')));
        if (newArray.length === 0) toast.dark("Can't find any movies with this name. Please try some other movies!", {autoClose: 3000,});
        dispatch(searchedItems(newArray));
    };
    const onEnter = (event) => {
        if (event.charCode === 13 || event.which === 13 || event.code === 'Enter') {
            searchData();
        }
    }
    useEffect(() => {
        if (handleBack) {
            setValue('');
            dispatch(removeSearchedItems())
        }
    }, [handleBack, dispatch])
return (
    <div className={'search_field'}>
        <input name="query" value={value} 
            placeholder="Search movies!" 
            onChange={(e) => getSearchQuery(e.target.value)}
            onKeyPress={(e) => onEnter(e)}
        />
        <button type="button" onClick={searchData}>
            <img className="search-icon" 
            src={process.env.PUBLIC_URL + '/assets/search.png'} 
            alt="Search" />
        </button>
        <ToastContainer />
    </div>
);
};

export default memo(Search);