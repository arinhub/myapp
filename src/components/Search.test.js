import renderer from 'react-test-renderer';
import React from 'react';
import Search from './Search';

const mockDispatchFunction = jest.fn();

jest.mock('react-redux', () => {
  return {
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn().mockImplementation(selector => selector()),
    useDispatch: () => {
      return mockDispatchFunction;
    },
  };
});

it('renders correctly', () => {
  const tree = renderer.create(
    <Search handleBack="false"/>
  );
  expect(tree).toMatchSnapshot();
});
