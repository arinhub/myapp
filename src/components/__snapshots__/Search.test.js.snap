// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="search_field"
>
  <input
    name="query"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Search movies!"
    value=""
  />
  <button
    onClick={[Function]}
    type="button"
  >
    <img
      alt="Search"
      className="search-icon"
      src="/assets/search.png"
    />
  </button>
  <div
    className="Toastify"
  />
</div>
`;
